gen/list.go

```diff
@@ -177,3 +181,45 @@ func (l *listGenerator) Equals(g Generator, spec *compile.ListSpec) (string, err
 
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
+
+func (l *listGenerator) zapMarshaler(
+	g Generator,
+	spec compile.TypeSpec,
+	root *compile.ListSpec,
+	fieldValue string,
+) (string, error) {
+	name := zapperName(g, spec)
```

Comment:
Are you sure you need to bother with the original `spec`? The wrapper is
generated with the root type regardless. The typedef-of-list can be cast to
the wrapper of the list directly.